org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NullAnnotationTests18.java

```diff
@@ -541,4 +541,28 @@ public void testBug576329() {
 		runner.expectedOutputString = "0";
 		runner.runConformTest();
 	}
+
+	@org.junit.Test
+	public void testNamedInstanceOfIsNonNull() {
+		Runner runner = getDefaultRunner();
+		runner.testFiles = new String[] {
+				"X.java",
+				"import org.eclipse.jdt.annotation.*;\n" +
+				"public class X {\n" +
+				"	public static void consumeNonNull(@NonNull String s) {\n" +
+				"		System.out.println(\"nonnull\");\n" +
+				"	}" +
+				"	public static void main(String... args) {\n" +
+				"		Object o = Math.random() > 0.5 ? new Object() : \"blah\";\n" +
```

Comment:
Please don't create a test that may randomly fail. (yes, I saw it fail)

Reply:
I still don't see the benefit of using Math.random() in tests, but at least id no longer causes harm.